pkg/utils/hostnameresolver.go

```diff
@@ -0,0 +1,15 @@
+package utils
+
+import "net"
+
+//go:generate mockery --name=HostnameResolver
+
+type HostnameResolver interface {
```

Comment:
Maybe we don't need a new file for this. This interface is only used by this gatherer, so we could have it directly in the gatherer file.
We already to this with other gatherers: https://github.com/trento-project/agent/blob/main/internal/factsengine/gatherers/systemd.go#L34

And you pass the `Resolver` or `DefaultResolver` from golang net package:
https://cs.opensource.google/go/go/+/master:src/net/lookup.go;l=128

`net.DefaultResolver{}` and use it as `r.LookupHost(context.Background(), host)`

Anyway, not a big deal

Reply:
I completely agree, and anyway I think it's not the place in package utils, If you want to put in a different file you can have it in the gatherers package, like we did with groups and users lookup for the dir-scan gatherer